integration-tests/testutils/build/build_test.go

```diff
@@ -263,3 +264,15 @@ func (s *BuildSuite) TestAssetsDoesNotBuildSnapdFromBranchIfNotInstructedTo(c *c
 		check.Commentf("Expected 0 call to execCommand with %s, got %d",
 			buildSnapdCmd, buildCall))
 }
+
+func (s *BuildSuite) TestAssetsHonoursBuildTags(c *check.C) {
+	testBuildTags := "mybuildtag"
+	Assets(s.useSnappyFromBranch, s.arch, testBuildTags)
```

Comment:
The amount of repetition here feels indicative that we can improve something. Also, the presence of the `"s."` prefix in the first or last parameters in only a few instances changes in a relevant way what the parameters mean. It's hard for the eye to spot these very small differences in a pattern that repeats equally most of the time.

Feels like this would be much better with a single struct parameter, which has proper defaults for everything. It then becomes much more obvious when you do, say:
```
BuildAssets(nil)
BuildAssets(&Config{BuildTags: testBuildTags})
```

That `Build` prefix behind the function name is also recommended, to make it more clear that this is actually what is going on. Assets is a noun, so I read this as a declaration such as "here are some assets".